### PR TITLE
Address static analysis warnings

### PR DESCRIPTION
**Pull request by mahilleb-msft**

### CODE REVIEW

File: adapters/tlsio_schannel.c [hunk 1]
```diff
@@ -746,7 +746,7 @@ static void on_underlying_io_bytes_received(void* context, const unsigned char*
                     {
                         LPVOID srcText = NULL;
                         if (FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, NULL,
-                            status, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), (LPTSTR)srcText, 0, NULL) > 0)
+                            status, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), (LPTSTR)&srcText, 0, NULL) > 0)
```
Comment (zhouwangzw) [inline adapters/tlsio_schannel.c:749]: this changes parameter from srcText to &srcText, by design?

File: src/base32.c [hunk 2]
```diff
@@ -322,7 +330,15 @@ char* Base32_Encode_Bytes(const unsigned char* source, size_t size)
     {
         /* Codes_SRS_BASE32_07_005: [ If size is 0 Base32_Encode shall return an empty string. ] */
         result = malloc(1);
-        strcpy(result, "");
+        if (result == NULL)
+        {
+            /* Codes_SRS_BASE32_07_014: [ Upon failure Base32_Encode_Bytes shall return NULL. ] */
+            LogError("encoding for zero-size input failed.");
```
Comment (zhouwangzw) [inline src/base32.c:336]: is this not malloc error?

**Review by zhouwangzw — APPROVED** [review]

File: src/base64.c [hunk 3]
```diff
@@ -242,7 +242,7 @@ static STRING_HANDLE Base64_Encode_Internal(const unsigned char* source, size_t
     neededSize += (size == 0) ? (0) : ((((size - 1) / 3) + 1) * 4);
     neededSize += 1; /*+1 because \0 at the end of the string*/
     /*Codes_SRS_BASE64_06_006: [If when allocating memory to produce the encoding a failure occurs then Base64_Encoder shall return NULL.]*/
-    encoded = (char*)malloc(neededSize);
+    encoded = (char*)malloc(neededSize + 1);
```
Comment (chlandsi) [inline src/base64.c:245]: isn't the +1 in line 243 enough?

File: adapters/tlsio_schannel.c (hunk 1 above)
Comment (mahilleb-msft) [reply adapters/tlsio_schannel.c:749]: Yes, this is a bug fix; with FORMAT_MESSAGE_ALLOCATE_BUFFER this parameters is a return parameter, returning the newly allocated buffer.

File: src/base64.c (hunk 3 above)
Comment (mahilleb-msft) [reply src/base64.c:245]: yes .. it's not to fix the error; however, this +1 is also present in upstream, so we decided to take it.

File: src/base32.c (hunk 2 above)
Comment (mahilleb-msft) [reply src/base32.c:336]: yes, it's a malloc error, but i was trying to align the error-log with the other ones in the same function.

**Review by chlandsi — APPROVED** [review]

**Review by fmegen — APPROVED** [review]

**Review by wolfma61 — APPROVED** [review]
:shipit:

**Pull request merged**